homeassistant/components/yardian/coordinator.py

```diff
@@ -0,0 +1,87 @@
+"""Update coordinators for Yardian."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import datetime
+import logging
+
+import async_timeout
+from pyyardian import AsyncYardianClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_WATERING_DURATION, DOMAIN, MANUFACTURER
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = datetime.timedelta(seconds=30)
+
+
+@dataclass
+class YardianDeviceState:
+    """Data retrieved from a Yardian device."""
+
+    zones: list[list]
+    active_zones: set[int]
+
+
+class YardianUpdateCoordinator(DataUpdateCoordinator[YardianDeviceState]):
+    """Coordinator for Yardian API calls."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        controller: AsyncYardianClient,
+    ) -> None:
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=entry.title,
+            update_method=self._async_update_data,
+            update_interval=SCAN_INTERVAL,
+        )
+
+        self._controller = controller
```

Comment:
```suggestion
        self.controller = controller
```

You don't need the property code below